ddtrace/tracer/tracer.go

```diff
@@ -371,3 +420,258 @@ func (t *tracer) sample(span *span) {
 	}
 	t.prioritySampling.apply(span)
 }
+
+type testStatsdClient struct {
+	mu          sync.RWMutex
+	gaugeCalls  []gaugeCall
+	incrCalls   []incrCall
+	countCalls  []countCall
+	timingCalls []timingCall
+	counts      map[string]int64
+	tags        []string
+	waitCh      chan struct{}
+	n           int
+	closed      bool
+}
+
+type gaugeCall struct {
+	name  string
+	value float64
+	tags  []string
+	rate  float64
+}
+
+type incrCall struct {
+	name string
+	tags []string
+	rate float64
+}
+
+type countCall struct {
+	name  string
+	value int64
+	tags  []string
+	rate  float64
+}
+
+type timingCall struct {
+	name  string
+	value time.Duration
+	tags  []string
+	rate  float64
+}
+
+func withStats(s statsdClient) StartOption {
```

Comment:
I think we should not make a habit of calling a _statsd client_ as _stats_ in the codebase.